sync CSS parsing tests and custom-elements state tests

keithamus opened the pull request:
#### 5a94eb22e5b63cb991524ec6e789aaba642a7fc4
<pre>
sync CSS parsing tests and custom-elements state tests
<a href="https://bugs.webkit.org/show_bug.cgi?id=264802">https://bugs.webkit.org/show_bug.cgi?id=264802</a>

Reviewed by Tim Nguyen.

This syncs the css/selectors/parsing, custom-elements/state, and
custom-state-pseudo-class WPTs, as well as removing the state.tentative
tests as they have were moved to state/tentative but prior sync failed
to remove this directory.

Additionally a lot of the css tests need rebaselining as the shared
assertions have changed.

Upstream commit: <a href="https://github.com/web-platform-tests/wpt/commit/bde41bd1d5581c22cb5a5ecb77ed2e366e8c18ac">https://github.com/web-platform-tests/wpt/commit/bde41bd1d5581c22cb5a5ecb77ed2e366e8c18ac</a>

* LayoutTests/imported/w3c/resources/import-expectations.json:
* LayoutTests/imported/w3c/web-platform-tests/css/css-scoping/slotted-parsing-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/selectors/parsing/parse-has-disallow-nesting-has-inside-has-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/selectors/parsing/parse-state-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/selectors/parsing/parse-state.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/support/parsing-testcommon.js:
* LayoutTests/imported/w3c/web-platform-tests/custom-elements/state.tentative/ElementInternals-states-expected.txt: Removed.
* LayoutTests/imported/w3c/web-platform-tests/custom-elements/state.tentative/ElementInternals-states.html: Removed.
* LayoutTests/imported/w3c/web-platform-tests/custom-elements/state.tentative/README.md: Removed.
* LayoutTests/imported/w3c/web-platform-tests/custom-elements/state.tentative/state-pseudo-class-expected.txt: Removed.
* LayoutTests/imported/w3c/web-platform-tests/custom-elements/state.tentative/state-pseudo-class.html: Removed.
* LayoutTests/imported/w3c/web-platform-tests/custom-elements/state.tentative/w3c-import.log: Removed.
* LayoutTests/imported/w3c/web-platform-tests/custom-state-pseudo-class/META.yml: Added.
* LayoutTests/imported/w3c/web-platform-tests/custom-state-pseudo-class/custom-state-set-strong-ref.tentative-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/custom-state-pseudo-class/custom-state-set-strong-ref.tentative.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/custom-state-pseudo-class/idlharness.window-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/custom-state-pseudo-class/idlharness.window.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/custom-state-pseudo-class/idlharness.window.js: Added.
(MyElement):
* LayoutTests/imported/w3c/web-platform-tests/custom-state-pseudo-class/state-css-selector.tentative-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/custom-state-pseudo-class/state-css-selector.tentative.html: Added.
* LayoutTests/platform/glib/imported/w3c/web-platform-tests/css/css-lists/parsing/list-style-computed.sub-expected.txt: Removed.
* LayoutTests/tests-options.json:

Canonical link: <a href="https://commits.webkit.org/270981@main">https://commits.webkit.org/270981@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/d76b6f404b68897362e933e71898c473344a7de5

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/26957 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/5573 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/28193 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/29168 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/24647 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/7430 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/2965 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/24517 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/27219 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/4397 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/23139 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/3847 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/3921 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/24137 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/29803 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/24602 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/24544 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/30132 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/3936 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/50/builds/2133 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/28043 "Passed tests") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/5395 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/23872 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/6480 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/4402 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/4305 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

nt1m reviewed (changes requested):
Can you avoid rebaselining tests that are unrelated to your WPT sync?

keithamus committed:
sync CSS parsing tests and custom-elements state tests
https://bugs.webkit.org/show_bug.cgi?id=264802

Reviewed by Tim Nguyen.

This syncs the css/selectors/parsing, custom-elements/state, and
custom-state-pseudo-class WPTs, as well as removing the state.tentative
tests as they have were moved to state/tentative but prior sync failed
to remove this directory.

Additionally a lot of the css tests need rebaselining as the shared
assertions have changed.

Upstream commit: https://github.com/web-platform-tests/wpt/commit/bde41bd1d5581c22cb5a5ecb77ed2e366e8c18ac

* LayoutTests/imported/w3c/resources/import-expectations.json:
* LayoutTests/imported/w3c/web-platform-tests/css/css-scoping/slotted-parsing-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/selectors/parsing/parse-has-disallow-nesting-has-inside-has-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/selectors/parsing/parse-state-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/selectors/parsing/parse-state.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/support/parsing-testcommon.js:
* LayoutTests/imported/w3c/web-platform-tests/custom-elements/state.tentative/ElementInternals-states-expected.txt: Removed.
* LayoutTests/imported/w3c/web-platform-tests/custom-elements/state.tentative/ElementInternals-states.html: Removed.
* LayoutTests/imported/w3c/web-platform-tests/custom-elements/state.tentative/README.md: Removed.
* LayoutTests/imported/w3c/web-platform-tests/custom-elements/state.tentative/state-pseudo-class-expected.txt: Removed.
* LayoutTests/imported/w3c/web-platform-tests/custom-elements/state.tentative/state-pseudo-class.html: Removed.
* LayoutTests/imported/w3c/web-platform-tests/custom-elements/state.tentative/w3c-import.log: Removed.
* LayoutTests/imported/w3c/web-platform-tests/custom-state-pseudo-class/META.yml: Added.
* LayoutTests/imported/w3c/web-platform-tests/custom-state-pseudo-class/custom-state-set-strong-ref.tentative-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/custom-state-pseudo-class/custom-state-set-strong-ref.tentative.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/custom-state-pseudo-class/idlharness.window-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/custom-state-pseudo-class/idlharness.window.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/custom-state-pseudo-class/idlharness.window.js: Added.
(MyElement):
* LayoutTests/imported/w3c/web-platform-tests/custom-state-pseudo-class/state-css-selector.tentative-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/custom-state-pseudo-class/state-css-selector.tentative.html: Added.
* LayoutTests/platform/glib/imported/w3c/web-platform-tests/css/css-lists/parsing/list-style-computed.sub-expected.txt: Removed.
* LayoutTests/tests-options.json:

Canonical link: https://commits.webkit.org/270981@main
